### PR TITLE
add ability to search for network interface by name

### PR DESCRIPTION
**Pull request by mdolian**
add the ability to search for a network interface by name using the tag: Name

### CODE REVIEW

**Comment by mdolian** [comment]
It looks like I have everything set except for the docs and I'm not really sure what I have to do.  Any guidance would be helpful.

File: lib/awspec/helper/finder/ec2.rb [hunk 1]
```diff
@@ -65,6 +65,11 @@ def find_network_interface(interface_id)
                                                          }
                                                        ]
                                                      })
+        resource = res.network_interfaces.single_resource(interface_id)
+        return resource if resource
+        res = ec2_client.describe_network_interfaces({
+                                                        filters: [{ name: 'tag:Name', values: [network-interface-id] }]
```
Comment (k1LoW) [inline lib/awspec/helper/finder/ec2.rb]: 1. `s/network-interface-id/interface_id/`

**Comment by k1LoW** [comment]
2. Prease run `bundle exec bin/toolbox docgen > doc/resource_types.md`.

Nice work!!

**Comment by mdolian** [comment]
I was just missing the 'bundle exec' part!  Running just 'bin/toolbox
docgen > doc/resource_types.md' works without an error but doesn't actually
do anything.

I think this tool is great.  I just started working at New York Life
Insurance and am on a team that is in charge of moving everything to AWS
over several years.  I plan on using this to ensure that all of our
CloudFormation templates do what we expect.  I'll likely be submitting more
pull requests as we use it more.

Thanks!
Matt

**Comment by mdolian** [comment]
It looks like I'm down to 1 failure.  The tests pass when searching for network_interface by name so I'm not really sure what is wrong.

**Comment by mdolian** [comment]
So I'm not sure what's going on but I can get the failing code to work in irb:

```
2.2.4 :004 > generator = Awspec::Generator::Doc::Type::NetworkInterface.new
 => #<Awspec::Generator::Doc::Type::NetworkInterface:0x007fac9193c598 @type_name="NetworkInterface", @type=network_interface 'eni-12ab3cde', @ret=#<struct Aws::EC2::Types::NetworkInterface network_interface_id="eni-12ab3cde", subnet_id="subnet-1234a567", vpc_id="vpc-ab123cde", availability_zone="ap-northeast-1c", description="", owner_id="1234567890", requester_id=nil, requester_managed=false, status="in-use", mac_address="00:11:aa:bb:cc:22", private_ip_address="10.0.1.1", private_dns_name=nil, source_dest_check=true, groups=[#<struct Aws::EC2::Types::GroupIdentifier group_name="my-security-group-name", group_id="sg-1a2b3cd4">], attachment=#<struct Aws::EC2::Types::NetworkInterfaceAttachment attachment_id="eni-attach-12ab3cde", instance_id="i-ec12345a", instance_owner_id="1234567890", device_index=0, status="attached", attach_time=nil, delete_on_termination=true>, association=nil, tag_set=[], private_ip_addresses=[#<struct Aws::EC2::Types::NetworkInterfacePrivateIpAddress private_ip_address="10.0.1.1", private_dns_name=nil, primary=true, association=nil>, #<struct Aws::EC2::Types::NetworkInterfacePrivateIpAddress private_ip_address="10.0.1.2", private_dns_name="", primary=false, association=nil>], interface_type=nil>, @matchers=["be_available, be_attaching, be_in_use, be_detaching", "belong_to_vpc", "belong_to_subnet"], @ignore_matchers=["be_available", "be_attaching", "be_in_use", "be_detaching"], @describes=[]> 
2.2.4 :005 > generator.generate_doc
 => "## <a name=\"network_interface\">network_interface</a>\n\nNetworkInterface resource type.\n\n### exist\n\n```ruby\ndescribe network_interface('eni-12ab3cde') do\n  it { should exist }\nend\n```\n\n\n### be_attached_to\n\n```ruby\ndescribe network_interface('eni-12ab3cde') do\n  it { should be_attached_to('my-ec2') }\n  it { should be_attached_to('my-ec2').as_eth0 }\nend\n```\n\n\n### be_available, be_attaching, be_in_use, be_detaching\n\n```ruby\ndescribe network_interface('eni-12ab3cde') do\n  it { should be_in_use }\nend\n```\n\n\n### have_private_ip_address\n\n```ruby\ndescribe network_interface('eni-12ab3cde') do\n  it { should have_private_ip_address('10.0.1.1').primary }\n  it { should have_private_ip_address('10.0.1.2') }\n  its(:private_ip_addresses_count) { should eq 2 }\nend\n```\n\n\n### have_security_group\n\n```ruby\ndescribe network_interface('eni-12ab3cde') do\n  it { should have_security_group('my-security-group-name') }\nend\n```\n\n\n### belong_to_subnet\n\n```ruby\ndescribe network_interface('eni-12ab3cde') do\n  it { should belong_to_subnet('my-subnet') }\nend\n```\n\n\n### belong_to_vpc\n\n```ruby\ndescribe network_interface('eni-12ab3cde') do\n  it { should belong_to_vpc('my-vpc') }\nend\n```\n\n### its(:network_interface_id), its(:subnet_id), its(:vpc_id), its(:availability_zone), its(:description), its(:owner_id), its(:requester_id), its(:requester_managed), its(:status), its(:mac_address), its(:private_ip_address), its(:private_dns_name), its(:source_dest_check), its(:association), its(:interface_type)\n" 
2.2.4 :006 > generated = generator.generate_doc
 => "## <a name=\"network_interface\">network_interface</a>\n\nNetworkInterface resource type.\n\n### exist\n\n```ruby\ndescribe network_interface('eni-12ab3cde') do\n  it { should exist }\nend\n```\n\n\n### exist\n\n```ruby\ndescribe network_interface('eni-12ab3cde') do\n  it { should exist }\nend\n```\n\n\n### be_attached_to\n\n```ruby\ndescribe network_interface('eni-12ab3cde') do\n  it { should be_attached_to('my-ec2') }\n  it { should be_attached_to('my-ec2').as_eth0 }\nend\n```\n\n\n### be_attached_to\n\n```ruby\ndescribe network_interface('eni-12ab3cde') do\n  it { should be_attached_to('my-ec2') }\n  it { should be_attached_to('my-ec2').as_eth0 }\nend\n```\n\n\n### be_available, be_attaching, be_in_use, be_detaching\n\n```ruby\ndescribe network_interface('eni-12ab3cde') do\n  it { should be_in_use }\nend\n```\n\n\n### have_private_ip_address\n\n```ruby\ndescribe network_interface('eni-12ab3cde') do\n  it { should have_private_ip_address('10.0.1.1').primary }\n  it { should have_private_ip_address('10.0.1.2') }\n  its(:private_ip_addresses_count) { should eq 2 }\nend\n```\n\n\n### have_private_ip_address\n\n```ruby\ndescribe network_interface('eni-12ab3cde') do\n  it { should have_private_ip_address('10.0.1.1').primary }\n  it { should have_private_ip_address('10.0.1.2') }\n  its(:private_ip_addresses_count) { should eq 2 }\nend\n```\n\n\n### have_security_group\n\n```ruby\ndescribe network_interface('eni-12ab3cde') do\n  it { should have_security_group('my-security-group-name') }\nend\n```\n\n\n### have_security_group\n\n```ruby\ndescribe network_interface('eni-12ab3cde') do\n  it { should have_security_group('my-security-group-name') }\nend\n```\n\n\n### belong_to_subnet\n\n```ruby\ndescribe network_interface('eni-12ab3cde') do\n  it { should belong_to_subnet('my-subnet') }\nend\n```\n\n\n### belong_to_vpc\n\n```ruby\ndescribe network_interface('eni-12ab3cde') do\n  it { should belong_to_vpc('my-vpc') }\nend\n```\n\n### its(:network_interface_id), its(:subnet_id), its(:vpc_id), its(:availability_zone), its(:description), its(:owner_id), its(:requester_id), its(:requester_managed), its(:status), its(:mac_address), its(:private_ip_address), its(:private_dns_name), its(:source_dest_check), its(:association), its(:interface_type), its(:network_interface_id), its(:subnet_id), its(:vpc_id), its(:availability_zone), its(:description), its(:owner_id), its(:requester_id), its(:requester_managed), its(:status), its(:mac_address), its(:private_ip_address), its(:private_dns_name), its(:source_dest_check), its(:association), its(:interface_type)\n" 
2.2.4 :007 > generated.include("network_interface")
NoMethodError: undefined method `include' for #<String:0x007fac918f4018>
	from (irb#1):7
2.2.4 :008 > generated.include?("network_interface")
 => true 
2.2.4 :009 > generated.include?("exist")
 => true 
```

File: lib/awspec/helper/finder/ec2.rb [hunk 2]
```diff
@@ -68,7 +68,7 @@ def find_network_interface(interface_id)
         resource = res.network_interfaces.single_resource(interface_id)
         return resource if resource
         res = ec2_client.describe_network_interfaces({
-                                                        filters: [{ name: 'tag:Name', values: [network-interface-id] }]
+                                                        filters: [{ name: 'tag:Name', values: [interface-id] }]
```
Comment (k1LoW) [inline lib/awspec/helper/finder/ec2.rb]: Do `s/interface-id/interface_id/` 👍

**Comment by mdolian** [comment]
Thanks for all the help!

**Comment by k1LoW** [comment]
@mdolian Yeah!!!! Great work!! Thank you!!! ありがとうございます！ 👍 👍 👍

**Comment by k1LoW** [comment]
LGTM!

**Comment by k1LoW** [comment]
Released as v0.57.0.

**Pull request merged**